app/controllers/application_controller.rb

```diff
@@ -1,2 +1,39 @@
 class ApplicationController < ActionController::API
+  private
+
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+
+    tenant = Tenant.find_by(:external_tenant => user_identity)
+    if tenant
+      set_current_tenant(tenant)
+    else
+      render :json => { :errors => "Unauthorized" }, :status => :unauthorized
+    end
+  end
+
+  def user_account_number
+    Base64.decode64(request.headers.fetch_path("x-rh-identity", "identity", "account_number"))
+  end
+
+  def user_identity
+    # x-rh-identity = {
+    #     "identity" => {
+    #         "account_number" => 123456,
+    #         "type" => "String"
+    #     },
+    #     "user" => {},
+    #     "system" => {},
+    #     "internal" => {},
+    # }
```

Comment:
I think the proper `identity` hash has `user`, `system` and `internal` keys inside the `identity` hash at the same level as `account_number` and `type`.

```
{
   "identity" : {
       "account_number": String,
       "type": String, // User or System
       "user": {}, // optional if User
       "system": {}, // optional if System
       "internal": {},
   }
}
```